cache/cache.go

```diff
@@ -57,3 +60,34 @@ func (c *Cache) WriteTo(w io.Writer) (int64, error) {
 	}
 	return total, nil
 }
+
+// Index generates an index for the given cache on a particular
+// column.  This is required for caches beyond a libnss-cache defined
+// size in order for them to be read correctly.
+func (c *Cache) Index(col int) bytes.Buffer {
+	var ordered []string
+	mapped := make(map[string]Entry, len(c.entries))
+	for i := range c.entries {
+		ordered = append(ordered, c.entries[i].Column(col))
```

Comment:
I would assign `c.entries[i].Column(col)` to a local variable to avoid calling `Column` twice.